integration_tests/src/main/python/conditionals_test.py

```diff
@@ -39,6 +40,23 @@ def test_if_else(data_gen):
                 'IF(a, b, {})'.format(null_lit),
                 'IF(a, {}, c)'.format(null_lit)))
 
+# Sadly you can not create literals for nested types in the form of sql string, according to
+# https://spark.apache.org/docs/latest/sql-ref-literals.html, and there is no API in Spark
```

Comment:
Is that really true?  These seem like cases where literals are being used:
```
scala> sql("select named_struct('a', 1, 'b', 2, 'c', 3, 'd', 4)").show
+------------------------------------+
|named_struct(a, 1, b, 2, c, 3, d, 4)|
+------------------------------------+
|                        [1, 2, 3, 4]|
+------------------------------------+


scala> sql("select named_struct('a', 1, 'b', 2, 'c', 3, 'd', 4)").printSchema
root
 |-- named_struct(a, 1, b, 2, c, 3, d, 4): struct (nullable = false)
 |    |-- a: integer (nullable = false)
 |    |-- b: integer (nullable = false)
 |    |-- c: integer (nullable = false)
 |    |-- d: integer (nullable = false)


scala> sql("select array(1, 2, 3, 4, 5)").show
+--------------------+
|array(1, 2, 3, 4, 5)|
+--------------------+
|     [1, 2, 3, 4, 5]|
+--------------------+


scala> sql("select array(1, 2, 3, 4, 5)").printSchema
root
 |-- array(1, 2, 3, 4, 5): array (nullable = false)
 |    |-- element: integer (containsNull = false)

scala> sql("select named_struct('a', array(1, 2, 3), 'b', 2, 'c', 'abc123')").show
+------------------------------------------------+
|named_struct(a, array(1, 2, 3), b, 2, c, abc123)|
+------------------------------------------------+
|                            [[1, 2, 3], 2, ab...|
+------------------------------------------------+


scala> sql("select named_struct('a', array(1, 2, 3), 'b', 2, 'c', 'abc123')").printSchema
root
 |-- named_struct(a, array(1, 2, 3), b, 2, c, abc123): struct (nullable = false)
 |    |-- a: array (nullable = false)
 |    |    |-- element: integer (containsNull = false)
 |    |-- b: integer (nullable = false)
 |    |-- c: string (nullable = false)
```

Reply:
Nice suggestion. Updated the tests accordingly.

For `NullGen`, only the top level one is handled as how it is doing, because I don't figure out how to convert a nested NullType to a SQL type string.